### PR TITLE
Fix get_max_num_running_seqs for waiting and swapped seq groups

### PR DESCRIPTION
**Pull request by zhuohan123**
This is the actual correct fix for the issues in #1034.

cc @Yard1

### CODE REVIEW

**Review by Yard1 — APPROVED** [review]
LGTM, thanks!

File: vllm/sequence.py [hunk 1]
```diff
@@ -250,8 +250,8 @@ def get_max_num_running_seqs(self) -> int:
                 # generation stage, we will have `best_of` sequences running.
                 return self.sampling_params.best_of
             # At sampling stages, return the number of actual sequences
-            # running.
-            return self.num_seqs(status=SequenceStatus.RUNNING)
+            # that are not finished yet.
+            return self.num_unfinished_seqs()
```
Comment (WoosukKwon) [inline vllm/sequence.py:254]: The semantic of the function will change because this also counts swapped sequences. Is this expected?

**Review by WoosukKwon — COMMENTED** [review]
Left minor comments.

File: vllm/sequence.py [hunk 2]
```diff
@@ -264,12 +264,21 @@ def get_seqs(
                 seq for seq in self.seqs_dict.values() if seq.status == status
             ]
 
+    def get_unfinished_seqs(self) -> List[Sequence]:
+        return [seq for seq in self.seqs_dict.values() if seq.is_finished()]
```
[the remark this answers is not in the document]
Comment (zhuohan123) [reply vllm/sequence.py:268]: Thanks for catching this! Sorry for the dumb mistake.

File: vllm/sequence.py (hunk 1 above)
Comment (zhuohan123) [reply vllm/sequence.py:254]: This is intended. `get_max_num_running_seqs` is being called for a swapped sequence when the swapped sequence is about to be swapped back to the running queue. This means these swapped sequences will soon become running sequences. This is the same logic as waiting sequences.

File: vllm/sequence.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (zhuohan123) [reply vllm/sequence.py]: Thanks for catching this! Sorry for the dumb mistake.

**Review by WoosukKwon — APPROVED** [review]
LGTM

**Pull request merged**